app/interactors/unset_current_map_if_deleting.rb

```diff
@@ -0,0 +1,12 @@
+class UnsetCurrentMapIfDeleting
```

Comment:
The interactor doesn't really have any knowledge of whether we are deleting, but the interactor is intended for performing a delete action. Perhaps something like `EnsureMapCanBeDeleted` or `PrepareMapForDeletion` expresses the behaviour in a clearer way?

Reply:
I think `PrepareMapForDeletion` is a better name and I'll change it to that. Thanks!